src/plugins/pbjx/components/history/delegate.js

```diff
@@ -51,8 +51,12 @@ const toFormValue = (id, value) => {
 
   let formValue = value;
 
-  if ((id.endsWith('refs') || ['channel_ref', 'sponsor_ref'].includes(id)) && !Array.isArray(value)) {
-    formValue = [NodeRef.fromString(value)];
+  if (id.endsWith('refs') || ['channel_ref', 'sponsor_ref'].includes(id)) {
```

Comment:
@rsumilang had a chat with Greg last night about simplifying the cms in several regards including forms, and we agree the logic in this delegate has gotten way too hairy. its a gnarly situation but we gotta do our part to keep it from getting any worse. i think we have to do the transformer strategy that we discussed in slack to minimize all these edge cases

Reply:
Are you talking about looking up the node to figure out the field types? If not, can you provide me more details.

Reply:
@rsumilang sure, its like we discussed before: there should be a transformer util for each field that can transform a field value into the form value and the form value into the field value. then these utils would be used by both the subscribers and this history jazz